presentation/src/main/java/com/smile/presentation/base/BaseViewModel.kt

```diff
@@ -22,8 +22,8 @@ abstract class BaseViewModel : ViewModel() {
                 + uiJob
     )
 
-    private val _failure: MutableLiveData<ErrorClass> = MutableLiveData()
-    val failure: LiveData<ErrorClass> = _failure
+    protected val failure: MutableLiveData<Exception> = MutableLiveData()
+    val failureLiveData: LiveData<Exception> = failure
```

Comment:
Nit: naming convention should be consistent. It will be best to append every livedata with `...LiveData` or not to at all. The `_` notation is also preferred for mutable properties though.